src/CallParser.ts

```diff
@@ -0,0 +1,177 @@
+import {
+	BlockStatement,
+	CallExpression,
+	FunctionDeclaration,
+	Identifier,
+	MemberExpression,
+	Node,
+	ReturnStatement,
+	UnaryExpression,
+} from "estree";
+import * as recast from "recast";
+
+import { parseComponentProps } from "./utils";
+
+class CallParser {
+	customJSXRuntime: FunctionDeclaration | null | undefined;
+
+	constructor(customJSXRuntime: FunctionDeclaration | null | undefined) {
+		this.customJSXRuntime = customJSXRuntime;
+	}
+
+	isJSXRuntime(node: Node): boolean {
+		return !!(
+			(node.type === "SequenceExpression" &&
+				((
+					(node.expressions[1]! as MemberExpression).property as Identifier
+				).name.endsWith("jsx") ||
+					(
+						(node.expressions[1]! as MemberExpression).property as Identifier
+					).name.endsWith("jsxs") ||
+					(this.customJSXRuntime &&
+						((node.expressions[1]! as MemberExpression).property as Identifier)
+							.name === this.customJSXRuntime.id!.name))) ||
+			(node.type === "MemberExpression" &&
+				((node.property as Identifier).name === "createElement" ||
+					(node.property as Identifier).name.endsWith("jsx") ||
+					(node.property as Identifier).name.endsWith("jsxs") ||
```

Comment:
`name` is not always a guaranteed property

```suggestion
					(node.property as Identifier).name?.endsWith("jsx") ||
					(node.property as Identifier).name?.endsWith("jsxs") ||
```